client/bootstrap

```diff
@@ -1334,36 +1344,38 @@ def execute_action(server, action_details, special_attr):
     action = action_details['action']
 
     description = ''
-    if 'description'in action_details:
+    if 'description' in action_details:
         description = '(%s)' % action_details['description']
 
-    if action not in sys.modules:
-        log('Downloading action %s%s' % (action, description))
-        filename = server.get_action(action)
-
     log('Executing action %s' % action)
     if 'onstart' in action_details:
         log('Action %s: %s' % (action, action_details['onstart']),
             xmpp=True)
 
     try:
-        if action in sys.modules:
-            module = sys.modules[action]
-        else:
-            module = imp.load_source(action, filename)
+        if action not in sys.modules:
+            log('Downloading action %s%s' % (action, description))
+            filename = server.get_action(action)
+
+            loader = importlib.machinery.SourceFileLoader(action, filename)
+            spec = importlib.util.spec_from_loader(loader.name, loader)
+            module = importlib.util.module_from_spec(spec)
+            sys.modules[action] = module
+            spec.loader.exec_module(module)
+
+        module = sys.modules[action]
 
         local_attr = action_details['attributes'] \
-                     if 'attributes' in action_details \
-                     else []
-        try:
-            module.main(Attributes(local_attr, special_attr))
-        except Exception as exc:
-            raise ZtpActionError(exc)
+            if 'attributes' in action_details \
+            else []
+
+        module.main(Attributes(local_attr, special_attr))
```

Comment:
why remove the try/except block here?

Reply:
block is embedded into another try/except that will raise the exception again into ZtpActionError.
So it looks to me that we are just masking the original exception with no real use.

Reply:
Gotcha.